modules/cache-common/src/main/java/org/opensearch/cache/common/tier/TieredSpilloverCache.java

```diff
@@ -257,19 +263,28 @@ public void put(ICacheKey<K> key, V value) {
             Tuple<V, String> cacheValueTuple = getValueFromTieredCache(true).apply(key);
             if (cacheValueTuple == null) {
                 // In case it is not present in any tier, put it inside onHeap cache by default.
-                try (ReleasableLock ignore = writeLock.acquire()) {
-                    onHeapCache.put(key, value);
+                if (evaluatePoliciesList(value, policies)) {
```

Comment:
This `evaluatePoliciesList` logic looks duplicated across `if-else`.
Better to check this upfront and return?

```
if (evaluatePoliciesList(value, policies)) {
   //
}
if (cacheValueTuple == null) {
   //
} else {
  //
}
```